include/vcmi/Entity.h

```diff
@@ -13,13 +13,23 @@
 VCMI_LIB_NAMESPACE_BEGIN
 
 class IBonusBearer;
+class FactionID;
+enum class ETerrainId;
+template<typename T> class Identifier;
 
 class DLL_LINKAGE WithBonuses
 {
 public:
 	virtual const IBonusBearer * getBonusBearer() const = 0;
 };
 
+class DLL_LINKAGE WithNativeTerrain
```

Comment:
IMO this class alongside with `WithBonuses` need better name...
Any ideas?

Reply:
What this name should be like?

Reply:
Not sure... Perhaps
BonusesProvider / BonusBearerProvider
NativeTerrainProvider
?

Might be better names, but can't think of any at the moment

Reply:
Another option
BonusBearerSource
NativeTerrainSource

XXXProvider might be better though

Reply:
If XXXProvider is not used anywhere in other parts of code? If not, I can rename. 

But honestly, I more like names like XXXInterface or WithXXX.

Reply:
I just feel that "WithBonuses" is incomplete name.
EntityWithBonuses is fine, but "WithBonuses" does not sounds right.

IEntityWithBonuses / IEntityWithNativeTerrain would also work IMO.

Not against XXXInterface form, but probably not suitable here.

Reply:
Renamed to IXXXProvider.